code/dal.py

```diff
@@ -0,0 +1,117 @@
+import MySQLdb
+import numpy as np
+import os
+
+
+def get_emails(num_emails=100, fetch_all=False):
+    """ Returns requested number of emails from a csv file. The function returns a numpy array of dimension (N,3)
+    where N is the number of rows, dimension1 is senderId, dimension2 is receiverIds, dimension3 is message body
+    If there are multiple csv files present, it selects the appropriate file that meets the <num_emails> requirement
+    and returns the requested number of rows
+    :param num_emails: The number of rows to be returned
+    :param fetch_all: Ignores num_emails & fetches all the emails in the enron dataset. The number of returned rows may
+    be very large """
+    file_path = '../data/'
+    if fetch_all:
+        file_name = 'all_emails.csv'
+        data = np.loadtxt(file_path + file_name, dtype='str', delimiter=',')
+        return data
+    else:
+        file_name = __get_appr_filename(num_emails, file_path)
+        data = np.loadtxt(file_path + file_name, dtype='str', delimiter=',')
+        if len(data) > num_emails:
+            return data[:num_emails]
+        else:
+            return data
+
+
+def load_from_db(num_emails=100, fetch_all=False):
+    """loads the requested number of emails from the database and dumps the file as csv. For this function to
+    run the mysql database must be set up locally.
+    if fetch_all is false, file is stored as <num_emails>_emails.csv
+    if the fetch_all is true, file is stored as all_emails.csv
+    :param num_emails: The number of emails to be fetched from the db
+    :param fetch_all: Will fetch all the emails from db. Large join query that may take some time"""
+    db_conn = MySQLdb.connect(host='localhost', user='root', passwd='mudit1990', db='EnronAHS')
```

Comment:
This is good enough for now.
In the future we can pull these from a constants file (gitignored)

Reply:
agreed